### PR TITLE
Support `copy` for TreeNodes

### PR DESCRIPTION
**Pull request by timholy**

### CODE REVIEW

**Review by c42f — APPROVED** [review]
Initially I thought the combination of shallow and deep copy was weird. But it's a reasonable analogy to how array copy copies the container but not the data. And it's pretty much how `copy(::Expr)` works too.  So LGTM :)

File: src/syntax_tree.jl [hunk 1]
```diff
@@ -244,6 +244,21 @@ function Base.push!(node::SN, child::SN) where SN<:AbstractSyntaxNode
     push!(args, child)
 end
 
+function Base.copy(node::TreeNode)
+    # copy is deep in the tree, shallow in the data
```
Comment (c42f) [inline src/syntax_tree.jl:248]:
```suggestion
    # copy the container but not the data (ie, deep copy the tree, shallow copy the data). copy(::Expr) is similar
```

**Comment by timholy** [comment]
OK if I tag a new release after this merges? (Or feel free to do so yourself.)

**Comment by c42f** [comment]
Go for it :+1:

**Pull request merged**